Dfe.Academies.Academisation.Domain/ProjectAggregate/Project.cs

```diff
@@ -68,12 +69,87 @@ public static CreateResult<IProject> Create(IApplication application, Establishm
 			//CapitalCarryForwardAtEndMarchNextYear = school.SchoolNFYCapitalForward.ConvertDeficitAmountToNegativeValue(school.SchoolNFYCapitalIsDeficit),
 			YearOneProjectedPupilNumbers = school.ProjectedPupilNumbersYear1,
 			YearTwoProjectedPupilNumbers = school.ProjectedPupilNumbersYear2,
-			YearThreeProjectedPupilNumbers	= school.ProjectedPupilNumbersYear3
+			YearThreeProjectedPupilNumbers = school.ProjectedPupilNumbersYear3
 		};
 
 		return new CreateSuccessResult<IProject>(new Project(projectDetails));
 	}
 
+	public CommandResult UpdatePatch(ProjectDetails detailsToUpdate)
+	{
+		Details = new ProjectDetails(1, 1)
+		{
+			HeadTeacherBoardDate = Details.HeadTeacherBoardDate == default(DateTime)
+			   ? null
+			   : detailsToUpdate.HeadTeacherBoardDate ?? Details.HeadTeacherBoardDate,
+			Author = detailsToUpdate.Author ?? Details.Author,
+			ClearedBy = detailsToUpdate.ClearedBy ?? Details.ClearedBy,
+			ProposedAcademyOpeningDate = detailsToUpdate.ProposedAcademyOpeningDate ?? Details.ProposedAcademyOpeningDate,
+			PublishedAdmissionNumber = detailsToUpdate.PublishedAdmissionNumber ?? Details.PublishedAdmissionNumber,
+			ViabilityIssues = detailsToUpdate.ViabilityIssues ?? Details.ViabilityIssues,
+			FinancialDeficit = detailsToUpdate.FinancialDeficit ?? Details.FinancialDeficit,
+			RationaleForProject = detailsToUpdate.RationaleForProject ?? Details.RationaleForProject,
+			RationaleForTrust = detailsToUpdate.RationaleForTrust ?? Details.RationaleForTrust,
+			RisksAndIssues = detailsToUpdate.RisksAndIssues ?? Details.RisksAndIssues,
+			RevenueCarryForwardAtEndMarchCurrentYear = detailsToUpdate.RevenueCarryForwardAtEndMarchCurrentYear == default(decimal)
+				? null
```

Comment:
What is the default value for `decimal`? I'm guessing that it's probably zero, in which case this code means that you can't use this method to set it back to zero _or_ null which seems unnecessarily restrictive. Would it be better to simplify so that you can't set it back to null but you could set it back to zero?

Reply:
I'm sure 0 would be the default. This is a copy from the trams repo. I see you're point though, will update it next.

Reply:
Just a thought before you do, better check that the front end uses nulls rather than defaults to signal an unchanged field. If not, I suppose we're stuck with it for now.

Reply:
The decimals are nullable on the front end.